### ISSUE TITLE
Error on starting: Quota limit exceeded, please wait and retry

### ISSUE THREAD

**Issue by sdaubert** [Valentin271/DeezerKodi#13]
**Describe the bug**  

I have just installed DeezerKodi 0.2.1, entered my creds (checked twice) and got this error: `Quota limit exceeded, please wait and retry`

**Environment (optional if log file provided) :**
- OS: OSMC
- Kodi version: 18.9

**Files**  

Here an extract from my log file:

```
2021-05-09 16:08:46.603 T:1565520096 WARNING: DeezerKodi: Failed to get token from file, trying API request instead
2021-05-09 16:09:53.138 T:1565520096   ERROR: DeezerKodi: Cannot get token from API
2021-05-09 16:09:58.406 T:1442832608   ERROR: GetDirectory - Error getting plugin://plugin.audio.deezer/
2021-05-09 16:09:58.406 T:1916621744   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.audio.deezer/) failed
2021-05-09 16:10:01.733 T:1281327328 WARNING: DeezerKodi: Failed to get token from file, trying API request instead
2021-05-09 16:10:01.936 T:1281327328   ERROR: DeezerKodi: Cannot get token from API
2021-05-09 16:10:10.211 T:1565520096   ERROR: GetDirectory - Error getting plugin://plugin.audio.deezer/
2021-05-09 16:10:10.211 T:1916621744   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.audio.deezer/) failed
2021-05-09 16:10:15.800 T:1281327328 WARNING: DeezerKodi: Failed to get token from file, trying API request instead
2021-05-09 16:10:16.198 T:1281327328   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.KeyError'>
                                            Error Contents: 'code'
                                            Traceback (most recent call last):
                                              File "/home/osmc/.kodi/addons/plugin.audio.deezer/addon.py", line 54, in <module>
                                                CONNECTION = Connection(ADDON.getSetting('username'), ADDON.getSetting('password'))
                                              File "/home/osmc/.kodi/addons/plugin.audio.deezer/resources/lib/DeezerApi/__init__.py", line 69, in __init__
                                                self._obtain_access_token()
                                              File "/home/osmc/.kodi/addons/plugin.audio.deezer/resources/lib/DeezerApi/__init__.py", line 131, in _obtain_access_token
                                                if 'error' in json_response and json_response['error']['code'] == QuotaException.CODE:
                                            KeyError: 'code'
                                            -->End of Python script error report<--
```

**Comment by uchi-mata** [comment]
Experiencing the same issue on Kodi 17.1.

**Comment by aminecmi** [comment]
Same here on Kodi matrix.

**Comment by lhcamilo** [comment]
Same on kodi 19.0 on ubuntu 21.04

**Comment by markiceman** [comment]
The same problem. Kodi 18.9., Android 7.1

**Comment by Valentin271** [comment]
Hello,

Do you have a premium account and is it a family or personal account ?
This happens sometimes, did anyone tried again later ?

I will try to reproduce that and make it more robust.

**Comment by aminecmi** [comment]
@Valentin271 I have a Hi/Fi account.

This happens every single time for me.

**Comment by sdaubert** [comment]
@Valentin271 I have a family account. This issue always happens.

**Comment by Valentin271** [comment]
@aminecmi HiFi account doesn't seems to be limited in the API.

I will try to edit a special debug version to get more informations as soon as possible.

**Comment by markiceman** [comment]
I have familly hifi account.

**Comment by lhcamilo** [comment]
I also have a Hi-fi account

**Comment by uchi-mata** [comment]
Deezer Premium here.

**Comment by scriptbee** [comment]
same error. deezer premium.

**Comment by airvb** [comment]
same same here libreelec last version family account

**Comment by teutat3s** [comment]
I'm seeing the same here, Deezer student account on Libreelec latest version `10.0.1`. Can we somehow help to debug this further?

**Comment by OraDotNetDev** [comment]
Same issue as OP, Premium family sub.
Attached two log files below, one with debug turned on.


**Environment (optional if log file provided) :**
- OS: OSMC (ARM 32-bit)
- Kodi version: 19.3 (Matrix)

**Files**  
[kodi.log](https://github.com/Valentin271/DeezerKodi/files/7960341/kodi.log)
[kodi.debug.log](https://github.com/Valentin271/DeezerKodi/files/7960534/kodi.debug.log)

**Comment by OraDotNetDev** [comment]
**Removing advancedsettings.xml** from ~/.kodi/userdata resolved this issue for me.

> Initially tried adding ` <cache><buffermode>0</buffermode>...` to advancedsettings.xml, which didn't work for me, but YMMV. It was a suggested fix, from [this post](https://forum.kodi.tv/showthread.php?tid=265356&pid=3083161#pid3083161) for resolving a similar _'invalid credentials error'_ for the Spotify addon. 

- _OS: OSMC_
- _Kodi: v 19.3_

```xml
<!-- contents of advancedsettings.xml -->
<advancedsettings>
  <cache>
   <memorysize>1073741824</memorysize>
   <readfactor>20</readfactor>
  </cache>
<!-- added disablehttp2 to try and resolve issue-->
  <network>
   <disablehttp2>true</disablehttp2><!--stutter problems on livestreams-->
  </network>
</advancedsettings>

```

**Comment by Valentin271** [comment]
Hello,

I've just released version 1.0.0 (after an incomfortably long amout of time I'll admit).
This bug **should** be resolved now.

If you still encounter this problem, please create another issue with fresh information.
